internal/exec/stages/disks/partitions.go

```diff
@@ -71,15 +71,31 @@ func (s stage) createPartitions(config types.Config) error {
 // also have non-zero start and size and non-nil start and size and label.
 // n.b. existing.{Size,Start}MiB must be converted to sectors first (yes the variable name becomes misleading)
 func partitionMatches(existing util.PartitionInfo, spec sgdisk.Partition) error {
+	if err := partitionMatchesCommon(existing, spec); err != nil {
+		return err
+	}
+	if spec.SizeInSectors != nil && *spec.SizeInSectors != existing.SizeInSectors {
+		return fmt.Errorf("size did not match (specified %d, got %d)", *spec.SizeInSectors, existing.SizeInSectors)
+	}
+	return nil
+}
+
+// partitionMatchesResize returns if the existing partition should be resized by evaluating if the
+// `resize`field is set to true and all of the specified attributes matches the existing partition's
+// attributes except the partition size.
+func partitionMatchesResize(existing util.PartitionInfo, spec sgdisk.Partition, resize bool) bool {
```

Comment:
The `resize` variable can be gathered from the `spec` parameter. Probably worth dropping that parameter and just doing the checks inside this function, e.x. remove the definition in `partitionDisk` & change this function to be:
```golang
func partitionMatchesResize(existing util.PartitionInfo, spec sgdisk.Partition) bool {
	return spec.Resize != nil && *spec.Resize && partitionMatchesCommon(existing, spec) == nil
}
```